docs/getting-started-step-by-step/00-overview.md

```diff
@@ -0,0 +1,44 @@
+# Overview Getting Started: Step-by-Step
+
+The following guide provides an easy, step-by-step path for developers and platform engineers to get familiar with Crossplane. It complements the current Getting Started guide, which focuses on getting an RDS Instance up and running quickly. In this example will use Object Storage as a simple example with quick turn around times.
+
+In this guide we are going to all important aspects of Crossplane: Managed Resources, Provider, ProviderConfigs, Compositions, CompositeResourceDefinitions (XRD), CompositeResources (XR), Claims and Configurations. 
+
+Here is the full architectural overview: 
+
+![architecture-overview](00-architecture-overview.jpg)
+
+Don't worry about the different terms. We will introduce them on our way.
+
+We will use this condensed form throughout the guide: 
+
+![architecture-overview-condensed](architecture-all.jpg)
+
+## Parts & Steps
+
+The guide consists of 4 parts with individual steps:
+
+* 1.) [Setup](01-setup.md): Setting up Crossplane on Kubernetes.
+    * Create a K8s Cluster
+    * Install the Crossplane CLI
+    * Install the Crossplane Runtime
+* 2a.) [Using Managed Resources from AWS](02a-managed-resources-aws.md): Connect Crossplane to manage AWS resources.
+    * Install the AWS Provider 
+    * Configure the AWS Provider
+    * Create, Update and Delete an S3 bucket
+* 2b.) Using Managed Resources from GCP: Connect Crossplane to manage GCP resources.
+    * Install the GCP Provider 
+    * Configure the GCP Provider
+    * Create, Update and Delete a Cloud Storage bucket
```

Comment:
Let's try to have all the examples cover all three "big" vendors (AWS, GCP, and Azure).

Reply:
So we actually used to have examples for the big vendors, but found that they got stale very quickly and reverted to providing examples for a single provider in order to avoid documentation staleness.

I'm all for adding more specific examples, but I'd like us to identify what we intend to do differently this time to avoid that happening again. One idea I've had in the past (that is probably difficult to pull off) is to have the parts of the documentation that pertain to a particular provider (e.g. AWS specific examples) live in the relevant repo near the code so that when folks add a field or promote a type from v1alpha1 to v1beta1 they can make the docs changes in the same PR.

Reply:
I agree we need to ship something supportable vs. something exhaustive but difficult to maintain. It'll defeat the purpose if the guides are out of date for each provider after we ship. 

Could we make sure to update the docs with every provider release?

Reply:
> Could we make sure to update the docs with every provider release?

This is the kind of maintenance burden I'd like to avoid - it puts the onus on folks releasing a provider to go find out everything that's changed (usually by other folks) and reflect those changes at release time. I'd prefer to optimise for the people making the changes (often community members) to be able to reflect their changes in the docs when they land their features; not super sure how to make this efficient though. It _seems_ like it should be easy to make a mental note like "we use RDS in the crossplane/crossplane docs, make sure to update those when the API changes" but history has proven that has not worked so well.

Reply:
I've added a task "How to make the examples in the docs maintainable" to this issue and would propose to discuss this outside of this PR